app/_hub/kong-inc/ai-semantic-cache/overview/_index.md

```diff
@@ -0,0 +1,156 @@
+---
+nav_title: Overview
+---
+
+## What is semantic caching?
+
+Semantic caching enhances data retrieval efficiency by focusing on the meaning or context of queries rather than just exact matches. It stores responses based on the underlying intent and semantic similarities between different queries and can then retrieve those cached queries when a similar request is made. For example, if a user asks, "how to integrate our API with a mobile app" and later asks "what are the steps for connecting our API to a smartphone application?", the LLM recognizes that both queries are seeking similar information about the steps to integrate an API with an app. It can then retrieve and reuse previously cached responses if they are contextually relevant, even if they are phrased differently. By leveraging semantic caching, the LLM can reduce redundant processing and speeds up the response times.
+
+When a new request is made, the system can retrieve and reuse previously cached responses if they are contextually relevant, even if the phrasing differs. This method reduces redundant processing, speeds up response times, and ensures that answers are more relevant to the user’s intent, ultimately improving overall system performance and user experience. 
+
+By using the AI Semantic Cache plugin, the plugin can handle the caching for you. You can also configure the vector database, caching mechanism, as well as embeddings.
+
+The AI Semantic Cache plugin might not be the best fit for you if following apply to you:
+* Storing semantic vectors or embeddings and performing similarity searches can require significant storage and computational resources. If you have limited hardware or budget constraints, this might be a concern.
+* If your application data isn't inherently semantic or exact matches are sufficient, the benefits of semantic caching might be minimal. Simple keyword-based or traditional caching might be more appropriate and efficient.
+
+## How it works
+
+{% mermaid %}
```

Comment:
This diagram is not very easy to read. Is there a better way to display this, and if not do we need the diagram?

Reply:
Maybe it's two diagrams?